Lib/tarfile.py

```diff
@@ -2232,6 +2232,8 @@ def makelink(self, tarinfo, targetpath):
         try:
             # For systems that support symbolic and hard links.
             if tarinfo.issym():
+                if os.path.lexists(targetpath):
```

Comment:
@JulienPalard, perhaps include a short comment here explaining why this is needed? Something along the lines of "tar should overwrite existing files with symlinks, but os.symlink raises an exception rather than overwriting."